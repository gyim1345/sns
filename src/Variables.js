export const post = {
  like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
  tag: [('boat', 'sky', 'landscape')],
  _id: '5e8ec52d9bb4913b5a844423',
  id: 1,
  title: 'posting with id 1',
  imageUrl: 'http://localhost:3000/static/images/1.jpg',
  userName: 'gibong@gmail.com'
};

export const posts = [
  {
    like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
    tag: [('boat', 'sky', 'landscape')],
    _id: '5e8ec52d9bb4913b5a844423',
    id: 1,
    title: 'posting with id 1',
    imageUrl: 'http://localhost:3000/static/images/1.jpg',
    userName: 'gibong@gmail.com'
  }
];

export const currentUser = 'gibong@gmail.com';

export const currentUserName = 'gibong'

export const comments = [
  {
    like: [('gibong@gmail.com', 'guy@gmail.com', 'noone@gmail.com')],
    _id: '5e90119435a2d94f9e08e35a',
    id: 1,
    postLId: 1,
    title: 'title1',
    userName: 'gibong@gmail.com',
    replyToCommentId: null
  }
];

export const info = {
  userURL: 'http://localhost:3000/static/images/user1.png',
  name: 'gibong@gmail.com',
  postCount: 16,
  followerNumber: 3,
  nickName: '기봉이',
  introductory: '나는 기봉이'
};

export const sizeOfPicture = {
  width: '10px',
  height: '10px'
};
