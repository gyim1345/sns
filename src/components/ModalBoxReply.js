import Modal from 'react-modal';
import React from 'react';
import { css } from '@emotion/core';

import CommentSvg from '../svgIcons/CommentSvg';
import Reply from './Reply';

function ModalBoxReply({
  commentAPI,
  indexOfCommentOnThisPosting,
  postings,
  currentUser,
  addComment
}) {
  const [modalIsOpen, setIsOpen] = React.useState(false);

  function closeModal() {
    setIsOpen(false);
  }

  return (
    <>
      <CommentSvg setIsOpen={setIsOpen} />
      <Modal isOpen={modalIsOpen} onRequestClose={closeModal} css={modalCss}>
        <div css={[outerBoxCss]}>
          <div css={[innerHeaderCss]}>Add Reply to Comment</div>
          <div>
            <Reply
              posting={postings}
              commentAPI={commentAPI}
              currentUser={currentUser}
              indexOfCommentOnThisPosting={indexOfCommentOnThisPosting}
              addComment={addComment}
              closeModal={closeModal}
            />
          </div>
        </div>
      </Modal>
    </>
  );
}

const outerBoxCss = css`
  display: flex;
  flex-direction: column;
`;

const innerHeaderCss = css`
  font-weight: bold;
  background-color: white;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.26);
  border-color: lightgray;
  margin-top: 0;
  padding: 10px;
  color: rgba(var(--i1d, 38, 38, 38), 1);
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica,
    Arial, sans-serif;
  font-size: 15px;
  line-height: 18px;
`;

const modalCss = css`
  position: absolute;
  top: 50%;
  left: 50%;
  right: auto;
  bottom: auto;
  border: 1px solid rgb(204, 204, 204);
  background: rgb(255, 255, 255);
  overflow: auto;
  border-radius: 12px;
  outline: none;
  margin-right: -50%;
  transform: translate(-50%, -50%);
  display: flex;
  flex-direction: column-reverse;
`;

const marginLeft0 = css`
  border: 1px solid rgba(var(--d0b, 219, 219, 219), 1);
  color: rgba(var(--f07, 38, 38, 38), 1);
  background-color: white;
  width: 300px;
  height: 48px;
  font-size: 14px;
  font-weight: bold;
`;

export default ModalBoxReply;
