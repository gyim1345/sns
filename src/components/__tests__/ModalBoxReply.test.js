import React from 'react';
import { mount } from 'enzyme';
import ModalBoxReply from '../ModalBoxReply';
import { comments, currentUser, posts } from '../../Variables';

describe('<ModalBoxReply />', () => {
  it('renders svg', () => {
    const wrapper = mount(
      <ModalBoxReply
        currentUser={currentUser}
        commentAPI={comments}
        indexOfCommentOnThisPosting={0}
        postings={posts}
      />
    );

    expect(wrapper.props().currentUser).toBe(currentUser);
    expect(wrapper.props().commentAPI).toBe(comments);
    expect(wrapper.props().indexOfCommentOnThisPosting).toBe(0);
    expect(wrapper.props().postings).toBe(posts);
    expect(wrapper.html()).toMatch('<svg')
    expect(wrapper).toMatchSnapshot();
  });
});
