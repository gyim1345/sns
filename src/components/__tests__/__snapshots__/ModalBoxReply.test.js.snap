// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalBoxReply /> renders svg 1`] = `
<ModalBoxReply
  commentAPI={
    Array [
      Object {
        "_id": "5e90119435a2d94f9e08e35a",
        "id": 1,
        "like": Array [
          "noone@gmail.com",
        ],
        "postLId": 1,
        "replyToCommentId": null,
        "title": "title1",
        "userName": "gibong@gmail.com",
      },
    ]
  }
  currentUser="gibong@gmail.com"
  indexOfCommentOnThisPosting={0}
  postings={
    Array [
      Object {
        "_id": "5e8ec52d9bb4913b5a844423",
        "id": 1,
        "imageUrl": "http://localhost:3000/static/images/1.jpg",
        "like": Array [
          "gibong@gmail.com",
        ],
        "tag": Array [
          "landscape",
        ],
        "title": "posting with id 1",
        "userName": "gibong@gmail.com",
      },
    ]
  }
>
  <CommentSvg
    setIsOpen={[Function]}
  >
    <svg
      aria-label="댓글 달기"
      className="_8-yf5 "
      fill="#262626"
      height="24"
      onClick={[Function]}
      viewBox="0 0 48 48"
      width="24"
    >
      <path
        clipRule="evenodd"
        d="M47.5 46.1l-2.8-11c1.8-3.3 2.8-7.1 2.8-11.1C47.5 11 37 .5 24 .5S.5 11 .5 24 11 47.5 24 47.5c4 0 7.8-1 11.1-2.8l11 2.8c.8.2 1.6-.6 1.4-1.4zm-3-22.1c0 4-1 7-2.6 10-.2.4-.3.9-.2 1.4l2.1 8.4-8.3-2.1c-.5-.1-1-.1-1.4.2-1.8 1-5.2 2.6-10 2.6-11.4 0-20.6-9.2-20.6-20.5S12.7 3.5 24 3.5 44.5 12.7 44.5 24z"
        fillRule="evenodd"
      />
    </svg>
  </CommentSvg>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="ModalBoxReply"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIk1vZGFsQm94UmVwbHkuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBNkRvQiIsImZpbGUiOiJNb2RhbEJveFJlcGx5LmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IE1vZGFsIGZyb20gJ3JlYWN0LW1vZGFsJztcbmltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcblxuaW1wb3J0IENvbW1lbnRTdmcgZnJvbSAnLi4vc3ZnSWNvbnMvQ29tbWVudFN2Zyc7XG5pbXBvcnQgUmVwbHkgZnJvbSAnLi9SZXBseSc7XG5cbmZ1bmN0aW9uIE1vZGFsQm94UmVwbHkoe1xuICBjb21tZW50QVBJLFxuICBpbmRleE9mQ29tbWVudE9uVGhpc1Bvc3RpbmcsXG4gIHBvc3RpbmdzLFxuICBjdXJyZW50VXNlcixcbiAgYWRkQ29tbWVudFxufSkge1xuICBjb25zdCBbbW9kYWxJc09wZW4sIHNldElzT3Blbl0gPSBSZWFjdC51c2VTdGF0ZShmYWxzZSk7XG5cbiAgZnVuY3Rpb24gY2xvc2VNb2RhbCgpIHtcbiAgICBzZXRJc09wZW4oZmFsc2UpO1xuICB9XG5cbiAgcmV0dXJuIChcbiAgICA8PlxuICAgICAgPENvbW1lbnRTdmcgc2V0SXNPcGVuPXtzZXRJc09wZW59IC8+XG4gICAgICA8TW9kYWwgaXNPcGVuPXttb2RhbElzT3Blbn0gb25SZXF1ZXN0Q2xvc2U9e2Nsb3NlTW9kYWx9IGNzcz17bW9kYWxDc3N9PlxuICAgICAgICA8ZGl2IGNzcz17W291dGVyQm94Q3NzXX0+XG4gICAgICAgICAgPGRpdiBjc3M9e1tpbm5lckhlYWRlckNzc119PuuMgOuMk+q4gCDri6zquLA8L2Rpdj5cbiAgICAgICAgICA8ZGl2PlxuICAgICAgICAgICAgPFJlcGx5XG4gICAgICAgICAgICAgIHBvc3Rpbmc9e3Bvc3RpbmdzfVxuICAgICAgICAgICAgICBjb21tZW50QVBJPXtjb21tZW50QVBJfVxuICAgICAgICAgICAgICBjdXJyZW50VXNlcj17Y3VycmVudFVzZXJ9XG4gICAgICAgICAgICAgIGluZGV4T2ZDb21tZW50T25UaGlzUG9zdGluZz17aW5kZXhPZkNvbW1lbnRPblRoaXNQb3N0aW5nfVxuICAgICAgICAgICAgICBhZGRDb21tZW50PXthZGRDb21tZW50fVxuICAgICAgICAgICAgICBjbG9zZU1vZGFsPXtjbG9zZU1vZGFsfVxuICAgICAgICAgICAgLz5cbiAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgPC9kaXY+XG4gICAgICA8L01vZGFsPlxuICAgIDwvPlxuICApO1xufVxuXG5jb25zdCBvdXRlckJveENzcyA9IGNzc2BcbiAgZGlzcGxheTogZmxleDtcbiAgZmxleC1kaXJlY3Rpb246IGNvbHVtbjtcbmA7XG5cbmNvbnN0IGlubmVySGVhZGVyQ3NzID0gY3NzYFxuICBmb250LXdlaWdodDogYm9sZDtcbiAgYmFja2dyb3VuZC1jb2xvcjogd2hpdGU7XG4gIGJveC1zaGFkb3c6IDAgMXB4IDJweCAwIHJnYmEoMCwgMCwgMCwgMC4yNik7XG4gIGJvcmRlci1jb2xvcjogbGlnaHRncmF5O1xuICBtYXJnaW4tdG9wOiAwO1xuICBwYWRkaW5nOiAxMHB4O1xuICBjb2xvcjogcmdiYSh2YXIoLS1pMWQsIDM4LCAzOCwgMzgpLCAxKTtcbiAgZm9udC1mYW1pbHk6IC1hcHBsZS1zeXN0ZW0sIEJsaW5rTWFjU3lzdGVtRm9udCwgJ1NlZ29lIFVJJywgUm9ib3RvLCBIZWx2ZXRpY2EsXG4gICAgQXJpYWwsIHNhbnMtc2VyaWY7XG4gIGZvbnQtc2l6ZTogMTVweDtcbiAgbGluZS1oZWlnaHQ6IDE4cHg7XG5gO1xuXG5jb25zdCBtb2RhbENzcyA9IGNzc2BcbiAgcG9zaXRpb246IGFic29sdXRlO1xuICB0b3A6IDUwJTtcbiAgbGVmdDogNTAlO1xuICByaWdodDogYXV0bztcbiAgYm90dG9tOiBhdXRvO1xuICBib3JkZXI6IDFweCBzb2xpZCByZ2IoMjA0LCAyMDQsIDIwNCk7XG4gIGJhY2tncm91bmQ6IHJnYigyNTUsIDI1NSwgMjU1KTtcbiAgb3ZlcmZsb3c6IGF1dG87XG4gIGJvcmRlci1yYWRpdXM6IDEycHg7XG4gIG91dGxpbmU6IG5vbmU7XG4gIG1hcmdpbi1yaWdodDogLTUwJTtcbiAgdHJhbnNmb3JtOiB0cmFuc2xhdGUoLTUwJSwgLTUwJSk7XG4gIGRpc3BsYXk6IGZsZXg7XG4gIGZsZXgtZGlyZWN0aW9uOiBjb2x1bW4tcmV2ZXJzZTtcbmA7XG5cbmNvbnN0IG1hcmdpbkxlZnQwID0gY3NzYFxuICBib3JkZXI6IDFweCBzb2xpZCByZ2JhKHZhcigtLWQwYiwgMjE5LCAyMTksIDIxOSksIDEpO1xuICBjb2xvcjogcmdiYSh2YXIoLS1mMDcsIDM4LCAzOCwgMzgpLCAxKTtcbiAgYmFja2dyb3VuZC1jb2xvcjogd2hpdGU7XG4gIHdpZHRoOiAzMDBweDtcbiAgaGVpZ2h0OiA0OHB4O1xuICBmb250LXNpemU6IDE0cHg7XG4gIGZvbnQtd2VpZ2h0OiBib2xkO1xuYDtcblxuZXhwb3J0IGRlZmF1bHQgTW9kYWxCb3hSZXBseTtcbiJdfQ== */",
        "name": "102pzl3-modalCss",
        "styles": "position:absolute;top:50%;left:50%;right:auto;bottom:auto;border:1px solid rgb(204,204,204);background:rgb(255,255,255);overflow:auto;border-radius:12px;outline:none;margin-right:-50%;transform:translate(-50%,-50%);display:flex;flex-direction:column-reverse;;label:modalCss;",
        "toString": [Function],
      }
    }
    isOpen={false}
    onRequestClose={[Function]}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="css-102pzl3-modalCss"
      closeTimeoutMS={0}
      isOpen={false}
      onRequestClose={[Function]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <Portal
        containerInfo={
          <div
            class="ReactModalPortal"
          />
        }
      >
        <ModalPortal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          className="css-102pzl3-modalCss"
          closeTimeoutMS={0}
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={false}
          onRequestClose={[Function]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {},
              "overlay": Object {},
            }
          }
        />
      </Portal>
    </Modal>
  </EmotionCssPropInternal>
</ModalBoxReply>
`;
